### ISSUE TITLE
Error when selecting using the parent axis when there is no parent node

### ISSUE THREAD

**Issue by mvanbutselaar** [FontoXML/fontoxpath#277]
This happens when using vanilla fontoxpath, which uses the `ExternalDomFacade` I think.

It throws an error saying `TypeError: Cannot read property 'nodeType' of null` in `getBucketsForNode()` with `null` as an argument, which is called by `getParentNode()` while there is no `parentNode`.

Cannot reproduce on the XPath playground, because that seems to use it's own `domFacade` implementation, but running the following code in node should trigger the error:
```javascript
const { evaluateXPathToFirstNode, evaluateXPathToNodes } = require("fontoxpath")
const { sync } = require('slimdom-sax-parser');

const xml = sync('<root><element /></root>');
const nodeHavingParentElement = evaluateXPathToNodes('//[parent::element()]', xml);
```

I think `ExternalDomFacade.getParentNode()` should check if there is a `parentNode`, and otherwise return `null`.

Also, the return value signature for `ExternalDomFacade.getParentNode()` is `Node` instead of `Node | null` even though it sometimes returns `null`, and `IDomFacade.getParentNode()` also supports returning `null`.

**Comment by DrRataplan** [comment]
Good find! I wonder how no one else ran into it!

**Comment by DrRataplan** [comment]
Found it and fixed it. As a sidenote, the XPath you wrote does not do what you expect it does: it is creating a sequence of arrays with a single value being the parent node of the node. I think you meant `//*[parent::element()]`.

Anyway, this is pushed to a branch, I'll publish it when the CI passed.

Thanks again!

**Comment by DrRataplan** [comment]
Just released 3.13.6 which fixes this bug

**Comment by mvanbutselaar** [comment]
The expression should probably have been like you stated, but it was a much simplified statement than in my actual use case. 

Thanks for for the quick response and fix.
